pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/QuestionDao.java

```diff
@@ -1843,13 +1844,42 @@ default Map<Long, QuestionBlockDef> collectBlockDefs(Collection<FormBlockDto> bl
             blockDef.setBlockGuid(blockDto.getGuid());
             blockDef.setShownExpr(blockDto.getShownExpr());
             blockDef.setEnabledExpr(blockDto.getEnabledExpr());
-
             blockDefs.put(blockDto.getId(), blockDef);
         }
 
         return blockDefs;
     }
 
+    default Map<Long, QuestionBlockDef> collectTabularBlockDefs(Collection<BlockTabularQuestionDto> blockDtos, long timestamp) {
```

Comment:
This code is pretty much same as above collectBlockDef. May be can be refactored !